### DISCUSSION TITLE
Testing API: Refresh Button doesn't surface a cancel?

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#688] [Extension Development QnA] [answered]
CC @connor4312 

I noticed that the `refreshHandler` hook provides a `CancellationToken`:

`refreshHandler: ((token: CancellationToken) => Thenable<void> | void) | undefined;`

However when starting a refresh, the cancel button doesn't appear, and I can't seem to figure out how to trigger this token.

I checked `test-provider-sample` and it doesn't have an example for using the cancellationToken in the refresh handler.

Am I missing something? When is the refreshHandler cancellationToken invoked?

**Comment by connor4312** [comment, marked as answer]
The refresh "cancel" button appears for me, replacing the refresh button while it's running.

![image](https://github.com/microsoft/vscode-discussions/assets/2230985/f4b024f9-69ee-442b-ab66-4e8e0327aa85)

**Reply by JustinGrote** [reply]
Thanks for confirming, I'll dig deeper then!
